org.eclipse.m2e.core/src/org/eclipse/m2e/core/project/IProjectConfiguration.java

```diff
@@ -62,4 +64,13 @@ private static List<String> parseProfiles(String profilesAsText, boolean status)
     return profiles;
   }
 
+  /**
+   * Gets the required Java version for executing Maven. If {@code null} is returned the project's build JRE or (if not
+   * available) the workspace default JRE should be used. Derived from the
+   * <a href="https://maven.apache.org/enforcer/enforcer-rules/requireJavaVersion.html">{@code maven-enforcer-plugin}'s
+   * {@code requireJavaVersion} rule</a>.
+   * 
+   * @return a version spec compliant with {@link VersionRange} spec
+   */
+  String getRequiredJavaVersion();
```

Comment:
Do you have a need for that as part of M2E's public API?
Actually I would consider this as internal.

In case there is a real need for that, I would rename it to `getExecutionJavaVersion()` or similar.

Reply:
Please don't use the `IProjectConfig` for this! This class is used to group project that have the same resolver context (e.g. selected properties and profiles) but not a general purpose place to store project specific settings, beside that these are workspace specific and cannot be shared.
This should be stored in the preferences (e.g. `org.eclipse.m2e.core.prefs` or even `org.eclipse.m2e.launching.prefs`).